### DISCUSSION TITLE
I need help with Python code, spreadsheet automation.

### DISCUSSION THREAD

**Discussion by Miguelbio012** [community/community#102007] [Programming Help] [answered]
### Body

I have code that connects to a web page (a store), and it takes information such as the name of the product and its price, it takes this information and creates a table with the names and prices of the matched products, but my problem is that the cell where these names and prices are located in the spreadsheet is wrong.

![Captura de tela 2024-01-25 095729](https://github.com/community/community/assets/138018833/6ced20a1-d65d-44ea-a98b-ee1bd1fa4dbb)

[code.txt](https://github.com/community/community/files/14052610/code.txt)

### Guidelines

- [X] I have read and understood this category's [guidelines](https://github.com/orgs/community/discussions/51384) before making this post.

**Comment by sbk2k1** [comment, marked as answer]
Hey @Miguelbio012 you can use the 'cell' attribute of the 'append' method. You can mention the row and column as parameters!

You can change your looping code as follows. I think this should work fine!

```
# Specify the starting cell for data
start_row = 2  # Starting from row 2 (A1 & B1 - for column indexes)

# Insert titles and prices into the sheet, starting from the specified cell
for titulo, preco in zip(titulos, precos):
    sheet_produtos.cell(row=start_row, column=1, value=titulo.text)
    sheet_produtos.cell(row=start_row, column=2, value=preco.text)
    start_row += 1
```

**Reply by Miguelbio012** [reply]
Sorry @sbk2k1, but I didn't have a solution, still the spreadsheet is placing information from cells A34, B34 to A65, B65.

**Reply by Miguelbio012** [reply]
![Captura de tela 2024-01-25 120517](https://github.com/community/community/assets/138018833/bc5bd030-80ad-4f3d-8ef2-7f3d611ac18c)

**Reply by sbk2k1** [reply]
Hey, @Miguelbio012  could you try to place some placeholders in case "titulo.txt" and "preco.txt" are not defined for some HTML elements? There may be some empty elements (32 of them specifically) with no text that are inserted in the first 32 rows and then the actual data starts from 34. Working with placeholders works correctly for me.

**Reply by sbk2k1** [reply]
And if that is the case, you can use any of the ways, 
1. Point out the divs with the data better. 
2. Add a check if blank spaces are being appended and prevent it (simple if-else)
3. Remove the first 32 data points If you know the structure remains the same.

**Reply by Miguelbio012** [reply]
It was a good idea! I'm trying to rephrase it here. Thanks.
